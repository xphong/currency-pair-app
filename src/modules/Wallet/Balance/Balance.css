.list {
  list-style: none;
  padding-left: 12px;
}

.currency {
  font-weight: 400;
  margin-right: 12px;
}