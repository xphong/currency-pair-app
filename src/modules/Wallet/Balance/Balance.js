import PropTypes from "prop-types";
import React, { Component } from "react";

import styles from './Balance.css';

export default class Balance extends Component {
  static propTypes = {
    balance: PropTypes.object
  };

  render() {
    const { balance } = this.props;

    if (!balance) {
      return <div>Unable to retrieve account balance</div>;
    }

    return (
      <div>
        <h2>Account Balance</h2>
        <ul className={styles.list}>
          {
            Object.keys(balance).map(key => (
              <li key={key}><span className={styles.currency}>{key}</span> {balance[key]}</li>
            ))
          }
        </ul>
      </div>
    );
  }
}
