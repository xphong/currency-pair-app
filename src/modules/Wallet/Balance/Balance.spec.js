import React from 'react';

import Balance from './Balance';

describe('Balance Component', () => {
  it('should render account balance', () => {
    const props = {
      balance: {
        USD: 1,
        BTC: 0
      }
    }
    const wrapper = shallow(<Balance {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render error message if no balance found', () => {
    const wrapper = shallow(<Balance />);

    expect(wrapper).toMatchSnapshot();
  });
});
