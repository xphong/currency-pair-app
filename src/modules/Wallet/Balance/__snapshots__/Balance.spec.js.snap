// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance Component should render account balance 1`] = `
<div>
  <h2>
    Account Balance
  </h2>
  <ul
    className="list"
  >
    <li
      key="USD"
    >
      <span
        className="currency"
      >
        USD
      </span>
       
      1
    </li>
    <li
      key="BTC"
    >
      <span
        className="currency"
      >
        BTC
      </span>
       
      0
    </li>
  </ul>
</div>
`;

exports[`Balance Component should render error message if no balance found 1`] = `
<div>
  Unable to retrieve account balance
</div>
`;
